### PR TITLE
.travis.yml: Run tests in docker

### PR DESCRIPTION
**Pull request by meetmangukiya**
# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,4 +22,3 @@ RUN addgroup -S errbot \
 
 USER errbot
 WORKDIR /opt/errbot
-CMD errbot
```
Comment (yukiisbored) [inline Dockerfile]: Why is this removed? `CMD` doesn't force you to run the command unlike `ENTRYPOINT`

File: .travis.yml [hunk 2]
```diff
@@ -25,6 +25,13 @@ script:
   - python -m pytest
   - docker build -t "corobo" .
   - docker images
+  - >
+    docker run -t -i corobo /bin/sh -c "
+        set -e -x
+        cd /opt/errbot;
+        pip install -r requirements.txt -r test-requirements.txt;
```
Comment (yukiisbored) [inline .travis.yml]: `requirements.txt` is already installed, just install `test-requirements.txt`

File: .travis.yml [hunk 3]
```diff
@@ -25,6 +25,13 @@ script:
   - python -m pytest
   - docker build -t "corobo" .
   - docker images
+  - >
+    docker run -t -i corobo /bin/sh -c "
```
Comment (yukiisbored) [inline .travis.yml]: Is there a reason why `-t` and `-i` is needed, there's no human interacting with the pseudo-tty and it's running inside a dumb terminal.

File: .travis.yml [hunk 4]
```diff
@@ -25,6 +25,13 @@ script:
...
+        set -e -x
+        cd /opt/errbot;
```
Comment (yukiisbored) [inline .travis.yml]: Redundant, We already set the work dir in the `Dockerfile` line 24

**Comment by meetmangukiya** [comment]
@rultor merge
